### PR TITLE
Show previous listing

### LINKED ISSUE #841: Feature request: show time between feedings in feeding list
It would be helpful if the feeding listing page `/feedings` showed for each feeding the time elapsed since the prior feeding. Right now it shows the date/time and duration of the feeding itself, but not time since the prior one.

image

tl;dr "Time since prior feeding" column after "Duration" column

### PR DESCRIPTION
**Pull request by shplack**
Add table column "Previous" to diaper change, feeding, pumping, sleep, and tummytime lists.

Closes #841

### CODE REVIEW

File: core/templates/core/diaperchange_list.html [hunk 1]
```diff
@@ -29,58 +29,68 @@ <h1>
...
+                                    <span title="{% trans "Solid" %}">💩</span>
+                                {% endif %}
+                            </td>
+                            <td>{{ change.get_color_display }}</td>
+                            <td>{{ change.amount|default_if_none:"" }}</td>
+                            <td>
+                                {% if prev_change %}
+                                    {% blocktrans trimmed with since=prev_change.time|timesince:change.time %}
+                                        {{ since }} ago
+                                    {% endblocktrans %}
```
Comment (cdubz) [inline core/templates/core/diaperchange_list.html:78]:
```suggestion
                                    {% blocktrans trimmed with time_ago=prev_change.time|timesince:change.time %}
                                        {{ time_ago }} ago
                                    {% endblocktrans %}
```

File: core/templates/core/feeding_list.html [hunk 2]
```diff
@@ -34,51 +34,61 @@ <h1>
...
+                            <td>{{ feeding.get_type_display }}</td>
+                            <td>
+                                {% if feeding.amount %}{{ feeding.amount }}{% endif %}
                             </td>
+                            <td>{{ feeding.duration|duration_string }}</td>
+                            <td>
+                                {% if prev_feeding %}
+                                    {% blocktrans trimmed with since=prev_feeding.start|timesince:feeding.start %}
+                                        {{ since }} ago
+                                    {% endblocktrans %}
```
Comment (cdubz) [inline core/templates/core/feeding_list.html:79]:
```suggestion
                                    {% blocktrans trimmed with time_ago=prev_feeding.start|timesince:feeding.start %}
                                        {{ time_ago }} ago
                                    {% endblocktrans %}
```

File: core/templates/core/pumping_list.html [hunk 3]
```diff
@@ -27,50 +27,60 @@ <h1>
...
+                                    <a href="{% url 'core:child' pumping.child.slug %}">{{ pumping.child }}</a>
+                                </td>
+                            {% endif %}
+                            <td>{{ pumping.amount }}</td>
+                            <td>{{ pumping.duration|duration_string }}</td>
+                            <td>
+                                {% if prev_pumping %}
+                                    {% blocktrans trimmed with since=prev_pumping.start|timesince:pumping.start %}
+                                        {{ since }} ago
+                                    {% endblocktrans %}
```
Comment (cdubz) [inline core/templates/core/pumping_list.html:68]:
```suggestion
                                    {% blocktrans trimmed with time_ago=prev_pumping.start|timesince:pumping.start %}
                                        {{ time_ago }} ago
                                    {% endblocktrans %}
```

File: core/templates/core/sleep_list.html [hunk 4]
```diff
@@ -27,50 +27,60 @@ <h1>
...
+                                <td>
+                                    <a href="{% url 'core:child' sleep.child.slug %}">{{ sleep.child }}</a>
+                                </td>
+                            {% endif %}
+                            <td>{{ sleep.duration|duration_string }}</td>
+                            <td>
+                                {% if prev_sleep %}
+                                    {% blocktrans trimmed with since=prev_sleep.start|timesince:sleep.start %}
+                                        {{ since }} ago
+                                    {% endblocktrans %}
```
Comment (cdubz) [inline core/templates/core/sleep_list.html:67]:
```suggestion
                                    {% blocktrans trimmed with time_ago=prev_sleep.start|timesince:sleep.start %}
                                        {{ time_ago }} ago
                                    {% endblocktrans %}
```

File: core/templates/core/tummytime_list.html [hunk 5]
```diff
@@ -26,45 +26,55 @@ <h1>
...
+                                <td>
+                                    <a href="{% url 'core:child' tummytime.child.slug %}">{{ tummytime.child }}</a>
+                                </td>
+                            {% endif %}
+                            <td>{{ tummytime.duration|duration_string }}</td>
+                            <td>
+                                {% if prev_tummytime %}
+                                    {% blocktrans trimmed with since=prev_tummytime.start|timesince:tummytime.start %}
+                                        {{ since }} ago
+                                    {% endblocktrans %}
```
Comment (cdubz) [inline core/templates/core/tummytime_list.html:68]:
```suggestion
                                    {% blocktrans trimmed with time_ago=prev_tummytime.start|timesince:tummytime.start %}
                                        {{ time_ago }} ago
                                    {% endblocktrans %}
```

**Review by cdubz — CHANGES_REQUESTED** [review]
This looks good! But please make the suggested changes to the `blocktrans` statements. This is for efficiency with the translation so it matches the same pattern already used and translated elsewhere.

Once those changes are in I'll merge this.

Thanks, again!

**Comment by shplack** [comment]
> This looks good! But please make the suggested changes to the `blocktrans` statements. This is for efficiency with the translation so it matches the same pattern already used and translated elsewhere.
> 
> Once those changes are in I'll merge this.
> 
> Thanks, again!

Done!

P.S.
I was following the syntax in [/core/templates/_timeline.html](https://github.com/babybuddy/babybuddy/blob/5bb88d2c85e6faa28eb56bd7602d6c239245f36f/core/templates/timeline/_timeline.html#L42)
Is that due for a change to `{{ time_ago }}`? I'm not too familiar with the codebase, just a shot in the dark.

**Review by cdubz — APPROVED** [review]

**Comment by cdubz** [comment]
All good now. Thanks, again!

> P.S.
> I was following the syntax in [/core/templates/_timeline.html](https://github.com/babybuddy/babybuddy/blob/5bb88d2c85e6faa28eb56bd7602d6c239245f36f/core/templates/timeline/_timeline.html#L42)
> Is that due for a change to `{{ time_ago }}`? I'm not too familiar with the codebase, just a shot in the dark.

The name of the variable is not the (only) important thing. Also the content of translatable string is relevant. Here I had you change this to `{{ time_ago }} ago` to match the same exact usage on [some dashboard cards](https://github.com/search?q=repo%3Ababybuddy%2Fbabybuddy%20%7B%7B%20time_ago%20%7D%7D%20ago&type=code).

While the example you were following is actually `{{ since }} ago ({{ time }})` -- so just a little different and those a different translatable string.

Really both of these use cases could probably be optimized to _just_ translate "ago" but that would be more refactoring than it's worth, I think.

**Pull request merged**